### PR TITLE
ciao-deploy: use a generic base clear image for building cnci.

### PR DESCRIPTION
**Pull request by kaccardi**
Rather than use a custom image, just use the latest clear release
as a base image and install the packages we need.

Signed-off-by: Kristen Carlson Accardi <kristen@linux.intel.com>

### CODE REVIEW

**Comment by kaccardi** [comment]
@rbradford my best guess on why this is failing on semaphore is the size of the image. We can try using a different clear base image - the cloud image is smaller, but would need the addition of "network-basic".
and "kvm_host" - https://download.clearlinux.org/current/config/image/cloud-config.json

**Comment by kaccardi** [comment]
and that image is qcow2

**Comment by rbradford** [comment]
@kaccardi  I think you need to install even more. I tried this on ccloudvm and i noticed that e.g. no dnsmasq was inside: here is the list of bundles for the original ciao-networking image: https://download.clearlinux.org/current/config/image/ciao-networking-config.json

**Comment by rbradford** [comment]
@kaccardi  also does the chroot swupd process work inside a proxied network?

**Comment by kaccardi** [comment]
@rbradford yes it should work on proxied network, but that's why we have to do that copy of /etc/resolv.conf before running swupd.

**Comment by kaccardi** [comment]
@rbradford I think this change will address your proxy issues. I also added code to clean up any old clear images that might be left if latest changes.

File: ciao-deploy/deploy/utils.go [hunk 1]
```diff
@@ -47,6 +47,31 @@ func DefaultImageCacheDir() string {
 	return path.Join(u.HomeDir, ".cache", "ciao", "images")
 }
 
+// CleanImage will remove all images from the cache directory that
+// match a specific pattern.
+func CleanImage(pattern string, keep []string, imageCacheDir string) error {
```
Comment (rbradford) [inline ciao-deploy/deploy/utils.go]: I think this would be better named CleanupImages

File: ciao-deploy/deploy/utils.go [hunk 2]
```diff
@@ -47,6 +47,31 @@ func DefaultImageCacheDir() string {
 	return path.Join(u.HomeDir, ".cache", "ciao", "images")
 }
 
+// CleanImage will remove all images from the cache directory that
+// match a specific pattern.
```
Comment (rbradford) [inline ciao-deploy/deploy/utils.go]: And can you mention what keep does?

File: ciao-deploy/deploy/cnci.go [hunk 3]
```diff
@@ -183,6 +241,27 @@ func prepareImage(ctx context.Context, baseImage string, agentCertPath string, c
 	return preparedImagePath, nil
 }
 
+func getCNCIURL() (string, error) {
```
Comment (rbradford) [inline ciao-deploy/deploy/cnci.go]: This should take a context.Context

File: ciao-deploy/deploy/cnci.go [hunk 4]
```diff
@@ -81,6 +82,27 @@ func unMountImage(ctx context.Context, devPath string, mntDir string) error {
 	return errOut
 }
 
+func getProxy(upper, lower string) (string, error) {
```
Comment (rbradford) [inline ciao-deploy/deploy/cnci.go]: I don't think it's worth taking these strings as parameters. Just hardcode them in this function.

**Pull request merged**